packages/@react-stately/data/src/useAsyncList.ts

```diff
@@ -199,6 +209,10 @@ function reducer<T, C>(data: AsyncListState<T, C>, action: Action<T, C>): AsyncL
             cursor: action.cursor
           };
         case 'error':
+          if (action.abortController !== data.abortController) {
```

Comment:
What was this case for?

Reply:
Paired with the added case below handling `loadingMore` if `loadingMore` is already happening. The change below will abort the newer (and extraneous) `loadingMore` actions which as a result will dispatch an error for the newer `loadingMore` actions. This case will then be triggered and preserve the current data state (i.e. we don't want to change the state to 'error' just cuz we canceled some extraneous loadMore calls)